packages/mottai_flutter_app/lib/auth/auth.dart

```diff
@@ -43,6 +60,50 @@ class AuthService {
   }) =>
       _auth.signInWithEmailAndPassword(email: email, password: password);
 
+  /// Googleでのサインイン
+  Future<UserCredential> signInWithGoogle({
+    required GoogleSignInAuthentication googleAuth,
+  }) async {
+    final credential = GoogleAuthProvider.credential(
+      accessToken: googleAuth.accessToken,
+      idToken: googleAuth.idToken,
+    );
+
+    final userCredential = await _auth.signInWithCredential(credential);
+    await _signIn(Authenticator.google, userCredential);
+    return userCredential;
+  }
+
+  /// Appleでのサインイン
+  Future<UserCredential> signInWithApple() async {
+    final appleProvider = AppleAuthProvider();
+    final UserCredential userCredential;
+    userCredential = await _auth.signInWithProvider(appleProvider);
+    await _signIn(Authenticator.apple, userCredential);
+    return userCredential;
```

Comment:
この処理が公式ドキュメント

https://firebase.flutter.dev/docs/auth/social/#apple

```dart
import 'dart:convert';
import 'dart:math';

import 'package:crypto/crypto.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:sign_in_with_apple/sign_in_with_apple.dart';

/// Generates a cryptographically secure random nonce, to be included in a
/// credential request.
String generateNonce([int length = 32]) {
  final charset =
      '0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyz-._';
  final random = Random.secure();
  return List.generate(length, (_) => charset[random.nextInt(charset.length)])
      .join();
}

/// Returns the sha256 hash of [input] in hex notation.
String sha256ofString(String input) {
  final bytes = utf8.encode(input);
  final digest = sha256.convert(bytes);
  return digest.toString();
}

Future<UserCredential> signInWithApple() async {
  // To prevent replay attacks with the credential returned from Apple, we
  // include a nonce in the credential request. When signing in with
  // Firebase, the nonce in the id token returned by Apple, is expected to
  // match the sha256 hash of `rawNonce`.
  final rawNonce = generateNonce();
  final nonce = sha256ofString(rawNonce);

  // Request credential for the currently signed in Apple account.
  final appleCredential = await SignInWithApple.getAppleIDCredential(
    scopes: [
      AppleIDAuthorizationScopes.email,
      AppleIDAuthorizationScopes.fullName,
    ],
    nonce: nonce,
  );

  // Create an `OAuthCredential` from the credential returned by Apple.
  final oauthCredential = OAuthProvider("apple.com").credential(
    idToken: appleCredential.identityToken,
    rawNonce: rawNonce,
  );

  // Sign in the user with Firebase. If the nonce we generated earlier does
  // not match the nonce in `appleCredential.identityToken`, sign in will fail.
  return await FirebaseAuth.instance.signInWithCredential(oauthCredential);
}
```

と一致していないのには理由がありますか？

Reply:
@KosukeSaigusa 
勘違いでしたらすみません。。
Androidで試したときに例外が発生してできなかったので、調べていたところ[こちらのissue](https://github.com/firebase/flutterfire/issues/2691)を見つけました。
上記Issueやほかのサイトなどを確認するとAndroidではこの方法が非対応みたいでしたので、別のログイン方法にしていました。
ただ、AndroidではAppleログインを行わないとのことでしたので、ソースはflutterfireを使用する方法に修正しました！

Reply:
Android での Apple ログインのためだったのですね！🙏私も知りませんでした、勉強になりました！
今回は Android で Apple ログインはしない前提でいきましょう！変更ありがとうございます！！